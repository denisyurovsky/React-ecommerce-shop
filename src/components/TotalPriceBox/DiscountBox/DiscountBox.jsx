import { Box } from '@mui/material';
import PropTypes from 'prop-types';
import React from 'react';

import { formatPrice } from '../../../helpers/formatData';

import styles from './DiscountBox.module.scss';

const DiscountBox = ({ sellersDiscount, personalDiscount }) => {
  const totalDiscount = sellersDiscount + personalDiscount;
  const isSellerDiscount = Boolean(sellersDiscount);
  const isPersonalDiscount = Boolean(personalDiscount);

  if (!isPersonalDiscount && !isSellerDiscount) {
    return null;
  }

  return (
    <Box className={styles.container}>
      {isPersonalDiscount && isSellerDiscount && (
        <p className={styles.title}>Discounts:</p>
      )}
      <table className={styles.table}>
        <tbody>
          {isSellerDiscount && (
            <tr className={styles.row}>
              <td>General discounts:</td>
              <td>-&nbsp;{formatPrice(sellersDiscount)}</td>
            </tr>
          )}
          {isPersonalDiscount && (
            <tr className={styles.row}>
              <td>Personal discounts:</td>
              <td>-&nbsp;{formatPrice(personalDiscount)}</td>
            </tr>
          )}
          {isPersonalDiscount && isSellerDiscount && (
            <tr className={styles.row}>
              <td>Total discounts:</td>
              <td>-&nbsp;{formatPrice(totalDiscount)}</td>
            </tr>
          )}
        </tbody>
      </table>
    </Box>
  );
};

DiscountBox.propTypes = {
  sellersDiscount: PropTypes.number,
  personalDiscount: PropTypes.number,
};

export default DiscountBox;
