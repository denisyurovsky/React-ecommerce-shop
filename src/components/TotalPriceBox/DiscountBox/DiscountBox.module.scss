@import "../../../assets/styles/colors.scss";

.container {
  font-size: 1.5em;
  margin-bottom: 10px;
}

.title {
  margin: 0;
  margin-bottom: 15px;
}

.table {
  border-spacing: 0;
  min-width: 230px;
  width: 100%;
  border: 0;

  & td {
    padding: 5px 0;
  }
}

.table .row {
  min-height: 20px;

  & td:nth-child(2) {
    padding-left: 25px;
  }
}

.table tr:last-child:not(:first-child) {
  & td {
    border-top: 1px dashed $greyDark;
    padding-top: 15px;
  }
}
