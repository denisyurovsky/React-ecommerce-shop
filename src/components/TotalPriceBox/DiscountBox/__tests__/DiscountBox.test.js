import { render } from '@testing-library/react';
import React from 'react';

import DiscountBox from '../DiscountBox';

describe('DiscountBox render correct snapshots', () => {
  const sellersDiscount = 1000;
  const personalDiscount = 1500;

  it('should render correct table with seller and personal discounts', () => {
    const { asFragment } = render(
      <DiscountBox
        sellersDiscount={sellersDiscount}
        personalDiscount={personalDiscount}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correct table with only seller discount', () => {
    const { asFragment } = render(
      <DiscountBox sellersDiscount={sellersDiscount} personalDiscount={0} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correct table with personal discount', () => {
    const { asFragment } = render(
      <DiscountBox sellersDiscount={0} personalDiscount={personalDiscount} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should return null if discounts equal to 0', () => {
    const { asFragment } = render(
      <DiscountBox sellersDiscount={0} personalDiscount={0} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
