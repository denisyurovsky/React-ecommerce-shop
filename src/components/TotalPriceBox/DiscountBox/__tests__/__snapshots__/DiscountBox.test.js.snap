// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscountBox render correct snapshots should render correct table with only seller discount 1`] = `
<DocumentFragment>
  <div
    class="container MuiBox-root css-0"
  >
    <table
      class="table"
    >
      <tbody>
        <tr
          class="row"
        >
          <td>
            General discounts:
          </td>
          <td>
            - 1 000 $
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`DiscountBox render correct snapshots should render correct table with personal discount 1`] = `
<DocumentFragment>
  <div
    class="container MuiBox-root css-0"
  >
    <table
      class="table"
    >
      <tbody>
        <tr
          class="row"
        >
          <td>
            Personal discounts:
          </td>
          <td>
            - 1 500 $
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`DiscountBox render correct snapshots should render correct table with seller and personal discounts 1`] = `
<DocumentFragment>
  <div
    class="container MuiBox-root css-0"
  >
    <p
      class="title"
    >
      Discounts:
    </p>
    <table
      class="table"
    >
      <tbody>
        <tr
          class="row"
        >
          <td>
            General discounts:
          </td>
          <td>
            - 1 000 $
          </td>
        </tr>
        <tr
          class="row"
        >
          <td>
            Personal discounts:
          </td>
          <td>
            - 1 500 $
          </td>
        </tr>
        <tr
          class="row"
        >
          <td>
            Total discounts:
          </td>
          <td>
            - 2 500 $
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`DiscountBox render correct snapshots should return null if discounts equal to 0 1`] = `<DocumentFragment />`;
